src/algos/base_class.py

```diff
@@ -339,6 +339,13 @@ def log_metrics(self, stats: Dict[str, Any], iteration: int) -> None:
                 imgs=stats["images"], key="sample_images", iteration=iteration
             )
 
+        #config["num_users"]
```

Comment:
So is the logic to log a graph at every round or only do it in a posthoc manner (in the plot utils) where we log the graph in the end? I am seeing code in both log_utils and plot_utils so not sure

Reply:
the logic is the log the global csv files for each round of each node. Once the final node of a round finishes, the plotter will work to plot the results of that round based on the data produced by the logger.

Reply:
But csv files were already getting logged about who is collaborating with whom at every round, why store separate csv files?

Reply:
In other words, there isn't any information that needs to be captured in addition to what we were already capturing for generating the three network visualizations (post training), correct?